src/main.rs

```diff
@@ -76,10 +79,61 @@ fn color_to_index(color: &[u8]) -> u8 {
     }).min_by_key(|&(_, diff)| diff).expect("4 components").0 as u8
 }
 
+fn get_target_json(url: &str) -> Result<TargetJson, Box<Error>> {
+    let new_target: serde_json::Value = reqwest::get(url)?.json()?;
+    let new_target = new_target.as_object().ok_or("Json format error")?;
+    macro_rules! tr {
+        ($e:expr) => {
+            ($e).ok_or("Json format error")?
+        }
+    }
+
+    let fallbacks = tr!(tr!(tr!(new_target.get("fallbacks")).as_array())
+        .iter()
+        .map(|x| x.as_str().map(|x| x.to_string()))
+        .collect::<Option<Vec<String>>>());
+
+    let new_target = TargetJson {
+        major_version: tr!(new_target.get("major_version"))
+                            .as_u64().map(|x| x as u32).unwrap_or(MAJOR_VERSION),
+        minor_version: tr!(new_target.get("minor_version"))
+                            .as_u64().map(|x| x as u32).unwrap_or(MINOR_VERSION),
+        x:             tr!(tr!(new_target.get("x")).as_u64()) as u32,
+        y:             tr!(tr!(new_target.get("y")).as_u64()) as u32,
+        image:         tr!(tr!(new_target.get("image")).as_str()).to_string(),
+        fallbacks:     fallbacks,
+    };
+
+    if new_target.major_version > MAJOR_VERSION {
+        println!("New major version is available. Must update!");
+        process::exit(1);
+    }
+
+    if new_target.minor_version > MINOR_VERSION {
+        println!("New minor version is available. Update when convenient.");
+    }
+
+    Ok(new_target)
+}
+
 fn main() {
-    let mut args = env::args().skip(1).fuse();
-    let mut username = args.next();
-    let mut password = args.next();
+    let matches = App::new("ruplace")
+        .arg(Arg::with_name("config")
+            .short("c")
+            .long("config")
+            .value_name("URL")
+            .help("Set a custom JSON config URL")
+            .takes_value(true))
+        .arg(Arg::with_name("username")
+            .required(false)
```

Comment:
The bot already "standardized" on two positional arguments, so I'd keep them as is.